nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -109,9 +109,8 @@ let
       ${config.systemd.package}/bin/systemctl --user start graphical-session.target
 
       # Allow the user to setup a custom session type.
-      if test -x ~/.xsession; then
-          exec ~/.xsession
-      fi
+      test -x ~/.nix-profile/bin/xsession && exec ~/.nix-profile/bin/xsession
```

Comment:
This probably should be documented. In addition, it might be a good idea to reuse a name that other distros use like Debian:

https://wiki.debian.org/Xsession

which uses:

- x-session-manager
- x-window-manager
- x-terminal-emulator

You can also use ```command -v xsession``` to allow any binary in PATH to be used.

Reply:
Agreed that more paths should be supported, in particular it should be possible to use `users.users.<name?>.packages` to install this command. Eventually I would like to move Home Manager to install packages through this option when used as a NixOS module.

Reply:
>Right now, if you have an xsession script then you can never select an alternative session in the DM, which is quite limiting.

I think it's the other way round — not using the xsession script in all cases would be more limiting! The xsession script can pass on to the desired session based on the relevant environment variable (I forget which one that is…), allowing users to set environment variables for their session for instance.

As for using $PATH, sure.

Reply:
Hmm, @lheckemann do you have an example of how to accomplish this in the xsession script? Based on the
https://github.com/NixOS/nixpkgs/blob/4889643c527b2efce1174b839250771bd8e62dbc/nixos/modules/services/x11/display-managers/default.nix#L115-L121
below I understand that the desired session is passed as an argument to the `xsession-wrapper` script and I don't see this argument being stored in an exported variable. It would be great if it was possible since I then could make Home Manager work with other session types without the user having to mess around with the HM `xsession.scriptPath` option.

Reply:
The DESKTOP_SESSION env var contains the path to the selected session script, as far as I understand.

Reply:
For me it seems to just show the name of the session, not the script path:
```console
$ echo $DESKTOP_SESSION 
home-manager
```
I also did a little survey in \#nixos and ottidmes and clever said their `$DESKTOP_SESSION` is set to `xfce+bspwm` and `xfce`, respectively.

At least on my system I don't see any environment variable pointing to the session script.

Reply:
Maybe that's only for xterm then… I get `/nix/store/qjhks5jpwwaq72pfrr5ghvb59nvm5h11-desktops/share/xsessions/xterm` because I didn't set up anything else since I use .xsession.